Automattic/wp-calypso
# Adds meatball menu to promote content in the Post and Page list

#### Proposed Changes

A new entry to "Promote Post" in the meatball menu on "Post List" and "Page List" entries. Once the user clicks on this new entry the "Promote a Post" widget will display on the screen with the post information of the selected post (or page). It has some similarities to the entry point on the Stats page to open this "Promote a Post" widget.

Post list:
![image](https://user-images.githubusercontent.com/43957544/183461999-2bd9d327-5431-4711-8b32-e270f9d0df8c.png)

Page list:
![image](https://user-images.githubusercontent.com/43957544/183469943-6f566a82-433e-4e96-8821-11dec615443e.png)

#### Testing Instructions

- Enable the feature flag promote-posts either on [startup](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags,reader%20yarn%20start) or [via url](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags%20via%20URLs,and%20disables%20feature%20bar.) using the ?flags=promote-post query string on the following urls
- Visit the "Post list" page. Click on the meatball menu and click on "Promote Post" button. 
![image](https://user-images.githubusercontent.com/43957544/183461519-823be279-d3eb-40b7-aafe-47d24a8ed0e7.png)

- The "Promote Post" widget should display in screen
- Only Posts and pages that are published will display this new entry menu.
- clicking the button should open the widget. However, this is not in production yet so without setting up your environment you can check the network tab and notice that it's trying to hit the script widget.js - this will repeatedly fail. In a working dev environment - it only ever loads the script once.
- This operation can be done also in "Pages List"

#### Pre-merge Checklist


Related to #
